rpcs3/Emu/Cell/PPUThread.cpp

```diff
@@ -2230,17 +2379,42 @@ extern void ppu_initialize()
 		prx_list.emplace_back(&prx);
 	});
 
+	// If empty we have no indication for cache state, check everything
+	bool compile_fw = prx_list.empty();
+
 	// Initialize preloaded libraries
 	for (auto ptr : prx_list)
 	{
-		ppu_initialize(*ptr);
+		compile_fw |= ppu_initialize(*ptr);
+	}
+
+	if (Emu.IsStopped())
+	{
+		return;
+	}
+
+	std::vector<std::string> dir_queue;
+
+	if (compile_fw)
+	{
+		const std::string firmware_sprx_path = vfs::get("/dev_flash/sys/external/");
+		dir_queue.emplace_back(firmware_sprx_path);
 	}
 
-	// Initialize SPU cache
-	spu_cache::initialize();
+	if (compiled_main)
```

Comment:
There is a problem with this check.
If I stop emulation during the last ppu_precompile phase, then compiled_main will always be true and we can't compile the mself files during regular boot anymore.

In my opinion we should always scan on boot.
But I guess boot times should be tested first.

Reply:
Maybe it's better to test it in production, evaluated to true always for now.

Reply:
Why compile main first then? That just doesn't make sense.

Reply:
Hmm, I'll try to make a function that checks if cache exists without compiling.

Reply:
Implemented.